src/CTA.FeatureDetection.Common/WCFConfigUtils/WCFBindingAndTransportUtil.cs

```diff
@@ -0,0 +1,320 @@
+                    bindingsTransportMap.Add(bindingName, new BindingConfiguration());
+                }
+            }
+        }
+
+        /// <summary>
+        /// Check for Binding and Mode in Code based WCF Service
+        /// </summary>
+        /// <param name="project">ProjectWorkspace object of Code based WCF Service</param>
+        /// <param name="bindingsTransportMap">Dictionary of binding and transport mode</param>
+        public static void CodeBasedCheck(ProjectWorkspace project, Dictionary<string, BindingConfiguration> bindingsTransportMap)
+        {
+            IEnumerable<InvocationExpression> addEnpointInvocations = project.GetInvocationExpressionsByMethodName(Constants.AddServiceEndpointType);
+
+            foreach (var addEnpointInvocation in addEnpointInvocations)
+            {
+                var argumentCount = addEnpointInvocation.Arguments.Count();
+
+                if (argumentCount == 1)
+                {
+                    var endpointIdentifier = addEnpointInvocation.Arguments.First();
+
+                    IEnumerable<ObjectCreationExpression> serviceEndpointObjectExpressions = project.GetObjectCreationExpressionBySemanticClassType(Constants.ServiceEndpointClass);
+
+                    var endpointArgumentObjects = serviceEndpointObjectExpressions.
+                        SelectMany(s => s.GetObjectCreationExpressionBySemanticNamespace(Constants.SystemServiceModelClass));
+
+                    var bindingArgumentObjects = endpointArgumentObjects.Where(e => e.SemanticClassType != Constants.EndpointAddressType);
+
+                    foreach(var binding in bindingArgumentObjects)
+                    {
+                        var bindingName = binding.SemanticClassType;
+                        var bindingNameFormatted = bindingName.ToString().ToLower();
+
+                        bindingsTransportMap.Add(bindingName, new BindingConfiguration
+                        {
+                            Mode = GetModeFromObjectArguments(binding.Arguments),
+                            EndpointAddress = GetEndpointFromInvocationArguments(addEnpointInvocation.Arguments)
+                        });
+                    }
+                }
+
+                var bindingArgument = addEnpointInvocation.Arguments.Where(a => a.SemanticType.ToLower().Contains("binding"));
+
+                var objectDeclarations = addEnpointInvocation.GetObjectCreationExpressionBySemanticNamespace(Constants.SystemServiceModelClass);
+
+                ObjectCreationExpression objectCreationExpression;
+
+                if(objectDeclarations.IsNullOrEmpty())
+                {
+                    if(!bindingArgument.IsNullOrEmpty())
+                    {
+                        var bindingName = bindingArgument.First().SemanticType;
+
+                        var objectCreationExpressionList = project.GetObjectCreationExpressionBySemanticClassType(bindingName);
```

Comment:
Can u check the performance on this? We invoke this twice (with different parameters). Wondering if we need to add an extension that accepts multiple parameters.

Reply:
I have not seen performance issues currently. We can discuss this further if there is an optimal way to do this.